### PR TITLE
Expose event logging downstream

### PR DESCRIPTION
**Pull request by nc6**
This PR exposes the event logging mechanism for use downstream, as well as making some slight changes to its interface.

No changes should be visible to clients using the existing interfaces.

### CODE REVIEW

**Review by goolord — APPROVED** [review]
lgtm

File: semantics/executable-spec/src/Control/State/Transition/Extended.hs [hunk 1]
```diff
@@ -253,6 +256,15 @@ type family EventReturnType ep sts a :: Type where
   EventReturnType 'EventPolicyReturn sts a = (a, [Event sts])
   EventReturnType _ _ a = a
 
+class EventReturnTypeRep ert where
```
Comment (goolord) [inline semantics/executable-spec/src/Control/State/Transition/Extended.hs:259]: maybe `EventReturnTypeRep` -> `ReifyEventPolicy` or something

File: semantics/executable-spec/src/Control/State/Transition/Extended.hs [hunk 2]
```diff
@@ -431,14 +462,12 @@ applySTS = applySTSOptsEither defaultOpts
     defaultOpts =
       ApplySTSOpts
         { asoAssertions = globalAssertionPolicy,
-          asoValidation = ValidateAll
+          asoValidation = ValidateAll,
+          asoEvents = EPDiscard
         }
 
-#ifdef STS_ASSERT
```
Comment (lehins) [inline semantics/executable-spec/src/Control/State/Transition/Extended.hs:437]: If `STS_ASSERT` flag is no longer used we might want to also remove it from the cabal file

File: semantics/executable-spec/src/Control/State/Transition/Extended.hs (hunk 1 above)
Comment (nc6) [reply semantics/executable-spec/src/Control/State/Transition/Extended.hs:259]: I went for consistency with `RuleTypeRep`, in the same file.

File: semantics/executable-spec/src/Control/State/Transition/Extended.hs (hunk 2 above)
Comment (nc6) [reply semantics/executable-spec/src/Control/State/Transition/Extended.hs:437]: Oop, good catch, not sure how this got deleted. Totally unintentional!

**Review by JaredCorduan — APPROVED** [review]
Thank you @nc6 , this looks great!

If I understand correctly, @DavidEichmann can now use `applyBlockOpts` with the right `EventReturnTypeRep` to make a version of `foldBlocks` that can be used to make a client that receives all the blocks, txs, together with the events. yea?

**Comment by nc6** [comment]
Yep, that's the idea!

**Pull request merged**